pkg/agents/team7/agent1/agent.go

```diff
@@ -0,0 +1,167 @@
+package team7agent1
+
+import (
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
+	"math/rand"
+)
+
+/*
+
+Openness 	high-> doesn't get effected by new floors, looks forward to new day
+			low-> does get effected by new floors
+
+Currently implemented by changing attributes on reshuffles
+
+Conscientiousness high-> plans ahead, attentive and takes into account messgages
+			low-> no planning, fails to complete assigned tasks
+
+Not implemented yet
+
+
+Extraversion high-> very likely to communicate, will share a lot of information
+			low-> does not like to communicate
+
+Not implemented yet
+
+Agreeableness high-> caring so more likely to go hungry for the betterment of others, trustworthy
+			low-> greedy, manipulative
+
+
+Implemented by intialising attributes, not considering health state yet
+
+
+Neuroticism high-> dramatic mood swings, struggles to recover after weak state
+			low-> stable, relaxed and resilient
+
+
+
+
+*/
+
+type team7Personalities struct {
+	openness int
+
+	conscientiousness int
+
+	extraversion int
+
+	agreeableness int
+
+	neuroticism int
+}
+
+type CustomAgent7 struct {
+	*infra.Base
+	personality  team7Personalities
+	greediness   int
+	kindness     int
+	daysHungry   int
+	seenPlatform bool
+	foodEaten    food.FoodType
+	prevHP       int
+	prevFloors   []int
+}
+
+func New(baseAgent *infra.Base) (infra.Agent, error) {
+	return &CustomAgent7{
+		Base: baseAgent,
+		personality: team7Personalities{
+			openness:          rand.Intn(100),
+			conscientiousness: rand.Intn(100),
+			extraversion:      rand.Intn(100),
+			agreeableness:     rand.Intn(100),
+			neuroticism:       rand.Intn(100),
+		},
+		greediness:   0,
+		kindness:     0,
+		daysHungry:   0,
+		seenPlatform: false,
+		foodEaten:    0,
+		prevHP:       100,
+		prevFloors:   []int{},
+	}, nil
+}
+
+func (a *CustomAgent7) Run() {
+
+	//initialise greediness and kindness
+	if a.Age() == 0 {
+		a.greediness = 100 - a.personality.agreeableness
+		a.kindness = a.personality.agreeableness
+	}
```

Comment:
Why would you not put this in the constructor?

Reply:
How do I use values that have just been initialised in a struct, 
basically: kindness = this.personality.agreeableness does not work

Reply:
See Slack message